tests/api/runtimes/test_mpijob.py

```diff
@@ -0,0 +1,121 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import typing
+import unittest.mock
+
+from kubernetes import client as k8s_client
+
+import mlrun.runtimes.pod
+from mlrun import code_to_function, mlconf
+from mlrun.api.utils.singletons.k8s import get_k8s
+from mlrun.runtimes.constants import MPIJobCRDVersions
+from tests.api.runtimes.base import TestRuntimeBase
+
+
+class TestMpiJob(TestRuntimeBase):
+    def custom_setup(self):
+        self.runtime_kind = "mpijob"
+        self.code_handler = "test_func"
+
+    def _get_worker_pod(self, phase="Running"):
+        return k8s_client.V1Pod(
+            metadata=k8s_client.V1ObjectMeta(
+                labels={
+                    "kind": "mpijob",
+                    "owner": "tester",
+                    "v3io_user": "tester",
+                    "mpijob": "v1/mpi-job-role=worker",
+                },
+                name=self.name,
+            ),
+            status=k8s_client.V1PodStatus(phase=phase),
+        )
+
+    def _get_launcher_pod(self, phase="Running"):
+        return k8s_client.V1Pod(
+            metadata=k8s_client.V1ObjectMeta(
+                labels={
+                    "kind": "mpijob",
+                    "owner": "tester",
+                    "v3io_user": "tester",
+                    "mpijob": "v1/mpi-job-role=launcher",
+                },
+                name=self.name,
+            ),
+            status=k8s_client.V1PodStatus(phase=phase),
+        )
+
+    def _mock_list_pods(self, workers=1, pods=None, phase="Running"):
+        if pods is None:
+            pods = [self._get_worker_pod(phase=phase)] * workers
+            pods += [self._get_launcher_pod(phase=phase)]
+        get_k8s().list_pods = unittest.mock.Mock(return_value=pods)
+
+    def _mock_get_namespaced_custom_object(self):
+        pass
+
+    def _generate_runtime(
+        self, kind=None, labels=None
+    ) -> typing.Union[mlrun.runtimes.MpiRuntimeV1, mlrun.runtimes.MpiRuntimeV1Alpha1]:
+        runtime = code_to_function(
+            name=self.name,
+            project=self.project,
+            filename=self.code_filename,
+            handler=self.code_handler,
+            kind=kind or self.runtime_kind,
+            image=self.image_name,
+            description="test mpijob",
+            labels=labels,
+        )
+        return runtime
+
+
+class TestMpiV1Runtime(TestMpiJob):
+    def custom_setup(self):
+        super(TestMpiV1Runtime, self).custom_setup()
+        self.name = "test-mpi-v1"
+        mlconf.mpijob_crd_version = MPIJobCRDVersions.v1
+
+    def custom_teardown(self):
+        pass
+
+    def _mock_get_namespaced_custom_object(self, workers=1):
```

Comment:
Can move this to `TestMpiJob` or just merge both classes as there is only one test here anyway.
Also move all the internal methods to be after the `test_` methods